src/plugins/gui/providers/moonpayProvider.ts

```diff
@@ -197,6 +197,24 @@ const NETWORK_CODE_PLUGINID_MAP: StringMap = {
   zksync: 'zksync'
 }
 
+// Special case memo creation for plugins
+// Memo type is not documented by Moonpay but can be inferred from /currencies response field "addressTagRegex"
+const createMemo = (pluginId: string, value: string): EdgeMemo => {
+  const memo: EdgeMemo = {
+    type: 'text',
+    value,
+    hidden: true
+  }
+
+  switch (pluginId) {
+    case 'ripple': {
+      memo.type = 'number'
+      memo.memoName = 'destination tag'
+    }    
+  }
+  return memo
```

Comment:
This code seems odd. Try:

```ts
const createMemo = (pluginId: string, value: string): EdgeMemo => {
  const memo: EdgeMemo = {
    type: 'text',
    value,
    hidden: true
  }

  switch (pluginId) {
    case 'ripple': 
      return {  ...memo, type: 'number' }
    default:
      return memo
  }
}
```

Now all the return statements live in the `switch`, and we avoid editing data. Also, I don't think `memoName` is needed in an EdgeSpendInfo?